code_generation/license.cpp

```diff
@@ -136,3 +137,32 @@ QString License::text() const
 
     return txt;
 }
+
+License::Type License::getLicenseNameByString(const QString &license_string)
+{
+    bool result;
+    QMetaEnum metaEnum = QMetaEnum::fromType<License::Type>();
+    License::Type license_enum = static_cast<License::Type>(
+            metaEnum.keyToValue(license_string.toStdString().c_str(), &result));
```

Comment:
`license_string.toLatin1().constData()` is more common for this, and likely more optimized.